frontend/dockerfile/dockerfile_test.go

```diff
@@ -255,6 +255,7 @@ func TestIntegration(t *testing.T) {
 		// Only use the amd64 digest,  regardless to the host platform
 		integration.WithMirroredImages(map[string]string{
 			"amd64/bullseye-20230109-slim:latest": "docker.io/amd64/debian:bullseye-20230109-slim@sha256:1acb06a0c31fb467eb8327ad361f1091ab265e0bf26d452dea45dcb0c0ea5e75",
+			"amd64/bullseye-20230109:latest":      "docker.io/amd64/debian:bullseye-20230109@sha256:640e07a7971e0c13eb14214421cf3d75407e0965b84430e08ec90c336537a2cf",
```

Comment:
Do we need to pull in a new image? Also, looks like it is not actually used by the test?

Reply:
It is needed and actually used for creating a dangling stage

Reply:
But that dangling stage should be completely ignored by the builder afaics. Like I can build a Dockerfile like this just fine:

```
from thereisnoimage as dangling
from alpine
```

Reply:
Fixed